iroh-bytes/src/store/mem.rs

```diff
@@ -512,155 +208,253 @@ impl super::Store for Store {
     }
 
     async fn create_tag(&self, hash: HashAndFormat) -> io::Result<Tag> {
-        let mut state = self.0.state.write().unwrap();
+        let mut state = self.write_lock();
         let tag = Tag::auto(SystemTime::now(), |x| state.tags.contains_key(x));
         state.tags.insert(tag.clone(), hash);
         Ok(tag)
     }
 
     fn temp_tag(&self, tag: HashAndFormat) -> TempTag {
-        TempTag::new(tag, Some(self.0.clone()))
+        TempTag::new(tag, Some(self.inner.clone()))
     }
 
-    async fn clear_live(&self) {
-        let mut state = self.0.state.write().unwrap();
-        state.live.clear();
-    }
-
-    fn add_live(&self, live: impl IntoIterator<Item = Hash>) -> impl Future<Output = ()> {
-        let mut state = self.0.state.write().unwrap();
-        state.live.extend(live);
-        futures::future::ready(())
-    }
-
-    fn is_live(&self, hash: &Hash) -> bool {
-        let state = self.0.state.read().unwrap();
-        // a blob is live if it is either in the live set, or it is temp tagged
-        state.live.contains(hash) || state.temp.contains(hash)
+    async fn gc_start(&self) -> io::Result<()> {
+        Ok(())
     }
 
     async fn delete(&self, hashes: Vec<Hash>) -> io::Result<()> {
-        let mut state = self.0.state.write().unwrap();
+        let mut state = self.write_lock();
```

Comment:
did you check that it is safe to use a sync mutex inside these async calls?

Reply:
This is the mem version. All the things it does are manipulating in memory set and maps protected by a rwlock. There isn't a single await anywhere. Async mutexes are only needed if the mutex spans an await point, right?

Reply:
and if the locks are only held for a few ms, as you still end up locking up a whole thread